TLM/TLM/Util/TrackUtils.cs

```diff
@@ -53,5 +57,39 @@ internal static class TrackUtils {
         /// false if vehicles going backward, bi-directional, or non-directional</returns>
         internal static bool IsGoingForward(this NetInfo.Lane laneInfo) =>
             laneInfo.m_finalDirection.IsGoingForward();
+
+        /// <summary>
+        /// Checks if the turning angle between two segments at the given node is within bounds.
+        /// </summary>
+        internal static bool CheckSegmentsTurnAngle(
+            ref NetSegment sourceSegment,
+            ref NetSegment targetSegment,
+            ushort nodeId) {
+            float turningAngle = 1f;
+            if (!targetSegment.m_overridePathFindDirectionLimit) {
+                turningAngle = 0.01f - Mathf.Min(sourceSegment.Info.m_maxTurnAngleCos, targetSegment.Info.m_maxTurnAngleCos);
+            }
+            if ((targetSegment.m_flags2 & NetSegment.Flags2.ForbidTurn) != 0 &&
+                !nodeId.ToNode().m_flags.IsFlagSet(NetNode.Flags.End)) {
+                turningAngle = -0.99f;
+            }
+            if (turningAngle < 1f) {
+                Vector3 sourceDirection = sourceSegment.GetDirection(nodeId);
+                Vector3 targetDirection = targetSegment.GetDirection(nodeId);
+                float dot = VectorUtils.DotXZ(sourceDirection, targetDirection);
+                return dot < turningAngle;
+            }
+
+            return true;
+        }
+
+        public static LaneEndTransitionGroup GetLaneEndTransitionGroup(VehicleInfo.VehicleType vehicleType) {
+            LaneEndTransitionGroup ret = 0;
+            if (vehicleType.IsFlagSet(LaneArrowManager.VEHICLE_TYPES))
+                ret |= LaneEndTransitionGroup.Car;
+            if (vehicleType.IsFlagSet(TrackUtils.VEHICLE_TYPES))
+                ret |= LaneEndTransitionGroup.Car;
```

Comment:
Should this be `Track`?

Reply:
Yeah I also think it's weird the const was defined there

Reply:
@kianzarrin are tracked lanes supposed to be setting `LaneEndTransitionGroup.Car` here?

Reply:
Yes. this is a bug. Will confuse VehicleAI.

Reply:
OK I fixed that but now 1053 is not fixed anymore.

Reply:
maybe before it appeared fixed because it was routing for cars